modules/querying/pages/modules/querying/pages/write-query-data-to-cloud.adoc

```diff
@@ -0,0 +1,57 @@
+= Write Query Output to Cloud
+
+Before you can write query results to an S3 bucket, you need to set up the connection to the S3 service. This requires credentials like an AWS Access Key ID and Secret Access Key. Make sure the necessary read/write permissions are granted for the S3 bucket.
+There are two methods to configure these credentials:
+
+== Using gadmin config
+
+This method configures the S3 connection for the entire TigerGraph cluster. Once you set the credentials, they apply to all users in the cluster. Here's how to do it:
+
+[source,GSQL]
+----
+gadmin config set GPE.QueryOutputS3AWSAccessKeyID <YOUR_AWS_ACCESS_KEY_ID>
+gadmin config set GPE.QueryOutputS3AWSSecretAccessKey <YOUR_AWS_SECRET_ACCESS_KEY>
+gadmin config apply -y
+gadmin restart gpe -y
+----
+* Replace <YOUR_AWS_ACCESS_KEY_ID> and <YOUR_AWS_SECRET_ACCESS_KEY> with your actual AWS credentials
+
+== Using GSQL Session Parameters 
+
+This method configures the credentials only for the current session and user.
+
+[source,GSQL]
+----
+SET s3_aws_access_key_id = <YOUR_AWS_ACCESS_KEY_ID>
+SET s3_aws_secret_access_key = <YOUR_AWS_SECRET_ACCESS_KEY>
+----
+* These parameters are set per session, so each user will need to provide their own credentials if they’re working with S3.
+* All queries run during this session will use these credentials to write to S3.
+
+[NOTE]:If both methods (using gadmin config and GSQL session parameters) are applied simultaneously, the GSQL session parameters take precedence.
+
+When you execute queries in TigerGraph that produce results, the query output can be written directly to the configured S3 bucket. Since S3 is a shared storage system, multiple nodes in a TigerGraph cluster might try to upload results to the same bucket at the same time.
+
+== Unique File Paths to Avoid Conflicts
+
+To avoid naming conflicts (multiple nodes trying to write to the same file), TigerGraph allows you to include a unique prefix in the file path. The path can include:
```

Comment:
"allows you to"
Is this an option that the user needs to make use of?  Does the user get to choose the suffix format?  It's not clear.  If the suffix format is fixed (predetermined), then don't say "allows you to".